docs/configuration/http_filters/cors_filter.rst

```diff
@@ -0,0 +1,60 @@
+.. _config_http_filters_cors:
+
+CORS filter
+====================
+
+This is a filter which handles CORS requests based on route or virtual host settings.
+For the meaning of the headers please refer to the pages below.
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS
+https://www.w3.org/TR/cors/
+
+
+.. code-block:: json
+
+  {
+    "type": "both",
+    "name": "cors",
+    "config": {}
+  }
+
+
+Settings
+--------
+
+Settings on a route take precedence over settings on the virtual host.
```

Comment:
That's surprising to me.  I would have assumed global config was overriden by per-cluster was overriden by per-host in most cases.  Unfortunately I'm having trouble telling the order from existing docs.  @mattklein123 can you confirm this is consistent?

Reply:
Maybe I mistook it. Here is how I understood it:

```
{
"virtual_host": {
...
    "cors": {},
    "routes": [{"prefix": "...", "cors": {...}}]
}
```

The CORS inside would take precedence. 

No problem to change it though. What I didn't like much about the description in the issue is that you have to enable CORS explicitly on the virtual_host and on the route because it says if either of them is off => it's off.
Maybe there is a better way?

Reply:
Rote should override virtual host. IMO if either of them is set it's on.

Reply:
Makes sense! Thanks.